aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -272,6 +286,57 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    // Offer rotation capability of this account to another address
+    public entry fun offer_rotation_capability_ed25519(account: &signer, rotation_capability_sig_bytes: vector<u8>, pk_bytes: vector<u8>, recipient_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(recipient_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+
+        let pubkey = ed25519::new_unvalidated_public_key_from_bytes(pk_bytes);
+        let rotation_capability_sig = ed25519::new_signature_from_bytes(rotation_capability_sig_bytes);
+
+        // Get the current authentication key of the account
+        let account_resource = borrow_global_mut<Account>(addr);
+        let curr_auth_key = create_address(account_resource.authentication_key);
+
+        // Check curr_auth_key matches expected_current_auth_key:
+        //  1. First, append the Ed25519 scheme identifier '0x00' to the PK
+        //  2. Second, hash this using SHA3-256
+        vector::push_back(&mut pk_bytes, 0);
+        let expected_current_auth_key = hash::sha3_256(pk_bytes);
+        assert!(create_address(expected_current_auth_key) == curr_auth_key, std::error::unauthenticated(EWRONG_CURRENT_PUBLIC_KEY));
```

Comment:
We're using this twice in this module. Can we define a function? Can it be public so other modules can use it?

Reply:
yea good point, will add a function for this